Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/InteractableInspector.cs

```diff
@@ -678,6 +699,66 @@ protected void SetupEventOptions()
             eventOptions = InteractableEvent.GetEventTypes();
         }
 
+        protected string[] GetEventList()
+        {
+            return new string[] { };
+        }
+
+        #endregion Events
+
+        #region PopupUtilities
+
+        /// <summary>
+        /// Look for speech commands in the MRTK Speech Command profile
+        /// </summary>
+        /// <param name="keywords"></param>
+        /// <returns></returns>
+        protected bool TryGetSpeechKeywords(out string[] keywords)
+        {
+            List<string> keys = new List<string>();
+            
+            if (MixedRealityToolkit.Instance && (MixedRealityToolkit.Instance.ActiveProfile.IsInputSystemEnabled ||
+                MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.SpeechCommandsProfile != null ||
+                MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.SpeechCommandsProfile.SpeechCommands.Length > 0))
+            {
+                int keywordCount = MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.SpeechCommandsProfile.SpeechCommands.Length;
+
+                for (var i = 0; i < keywordCount; i++)
+                {
+                    keys.Add(MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile.SpeechCommandsProfile.SpeechCommands[i].Keyword);
+                }
+
+                keys.Insert(0, "None");
+                keywords = keys.ToArray();
+
+                return true;
+            }
+            else
+            {
+                keywords = null;
+                return false;
+            }
+        }
+        
+        /// <summary>
+        /// Get the index of the speech keyword array item based on it's name, pop-up field helper
+        /// </summary>
+        /// <param name="option"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public int KeywordLookup(string option, string[] options)
```

Comment:
You can delete this method and just use [Array.IndexOf](https://docs.microsoft.com/en-us/dotnet/api/system.array.indexof?redirectedfrom=MSDN&view=netframework-4.8#System_Array_IndexOf_System_Array_System_Object_) instead. `Array.IndexOf(options, option);`. Also, any reason you would want this to be public?

Reply:
There's a slight difference here than Array.IndexOf. This starts at item one in the array because "None" was added at item zero. If the speech command is not in the list, I want it to be zero, which is "None".

I was concerned that someone could create a speech command called "None" for some scenario, like choosing the amount of something via speech command. In that situation, a created speech command would match the inserted "None" value at index zero and turn off  listening for voice commands. It would be weird that there would be two "None" values in the list, but that's an extreme edge case that would not break anything.

I wanted an easy way to turn off speech command handling without adding another toggle to enable/disable it. But any value we insert could potentially be a speech command that someone could create. If this is too weird, I can put in the toggle :)

Reply:
I see. Up to you. Just wanted to make sure you were aware of the tools available. If you wanted to use LINQ you could also do keywords.Skip(1).IndexOf(...) or similar to skip the first element. 

Is it a fair assumption to always assume that the first element is keywords.None? If "None" is not even a valid keyword, why is it even in the keywords list?

It's not a huge deal, just wondering if I can provide some suggestions that could help simplify the code. It was not obvious to me that the first element in the list always represents "no keyword". I would expect if no keywords are registered for the keywords list to be empty.

Reply:
Thanks for the suggestions, I don't use LINQ very much and should learn more about it.

Leaving the first item blank or just spaces does not work in the editor popup, it just ignores the item without valid characters in the drop down so it doesn't should up as an option. I replaced None with -No Selection- as the first item, it's more clear and gives the user the feeling that they can select it as an option to turn the feature off.